### ISSUE TITLE
[BUG] 通过transformers对base model进行全量微调时loss归零

### ISSUE THREAD

**Issue by Sniper970119** [QwenLM/Qwen#967]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

在全量微调qwen-7b时，loss在几个step内迅速归零。具体归零的step数并不固定，但是在前10step基本都会出现。

<img width="356" alt="image" src="https://github.com/QwenLM/Qwen/assets/30463691/f451c20a-b19c-41c1-98be-5af4c67f03bb">
梯度在达到一个数千的峰值后也归零。
<img width="348" alt="image" src="https://github.com/QwenLM/Qwen/assets/30463691/f8d0b455-a9cb-4da2-878b-5540f5e8ea28">

已经重新下载过预训练权重。

### 期望行为 | Expected Behavior

loss、grad正常。

另外，我们还跑了1.8B的qwen，loss和grad都是正常的，应该可以排除代码和数据的异常。

<img width="707" alt="image" src="https://github.com/QwenLM/Qwen/assets/30463691/54f9bc3f-5a14-4d6b-809e-81100c3b4ba1">

### 运行环境 | Environment

```Markdown
- OS: Linux
- Python: 3.8
- Transformers: 3.35.0
- PyTorch: 2.0,1
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):11.7
```

**Comment by Sniper970119** [comment]
又跑了一下，似乎不是异常梯度的问题。
在第二个step loss就归零，同时梯度一直不变
<img width="707" alt="image" src="https://github.com/QwenLM/Qwen/assets/30463691/511332f8-bc30-4eba-949c-c224018b349b">

**Comment by jklj077** [comment]
> transformers: 3.35.0 

请先更新下transformers看看

**Comment by Sniper970119** [comment]
更新到了trms 4.36.2还是不行。单独一提，由于`modeling_qwen.py`版本是trms4.34.0，`gradient_checkpointing`在4.35+的支持有变化，我给`modeling_qwen.py`中大约760+行的`_set_gradient_checkpointing`进行了注释。

使用bf16、max_length 8192进行训练。

<img width="352" alt="image" src="https://github.com/QwenLM/Qwen/assets/30463691/61c60201-884a-4e25-b338-e675c5ddd1e0">

**Comment by Sniper970119** [comment]
已经知道具体原因了。。删掉deepspeed的`communication_data_type=fp32`配置就可以了，不过能请教一下为什么会发生这样的问题吗？默认是fp16，fp16转fp32不是无损的吗？

**Comment by jklj077** [comment]
不太清楚了，原因建议咨询DeepSpeed哈。这个配置本repo里的ds_config应该都没有，deepspeed会根据设定自己选择。本issue先关掉了，如果是Qwen代码的问题，请重开吧，我们来修正（1.8B如果开这个正常的话，感觉跟Qwen代码关系比较低）。
